### ISSUE TITLE
Request timeout: /distributions/gradle-8.7-rc-3-wrapper.jar.sha256

### ISSUE THREAD

**Issue by Omico** [gradle/wrapper-validation-action#191]
![image](https://github.com/gradle/wrapper-validation-action/assets/14316223/f063bf66-b591-41cc-9bf9-063bbce1d656)

**Comment by JLLeitschuh** [comment]
@bigdaz is there a potential that this occurred during an RC release? Perhaps there's some race condition between the `JSON` and the `sha256` checksum existing in the AWS bucket?

**Comment by Omico** [comment]
I guess if we merge #186, the problem can be solved.

**Comment by bigdaz** [comment]
@Omico Do you see this error consistently, or did it resolve itself?
You're right that a new release with the latest checksums would have prevented this failure, but the fetch-on-demand functionality should continue to work for newly released Gradle versions.

**Comment by JLLeitschuh** [comment]
> You're right that a new release with the latest checksums would have prevented this failure, but the fetch-on-demand functionality should continue to work for newly released Gradle versions.

Sorry, to clarify, do you know if the Gradle release process uploads the `https://services.gradle.org/distributions/gradle-[VERSION]-wrapper.jar.sha256` first, or updates the `https://services.gradle.org/versions/all` first.

Depending upon ordering in the release process, there could be a race condition where the `wrapper-validation-action` has pulled the `https://services.gradle.org/versions/all`, but the `https://services.gradle.org/distributions/gradle-[VERSION]-wrapper.jar.sha256` doesn't exist yet.

**Comment by bigdaz** [comment]
> Sorry, to clarify, do you know if the Gradle release process uploads the https://services.gradle.org/distributions/gradle-[VERSION]-wrapper.jar.sha256 first, or updates the https://services.gradle.org/versions/all first.

I don't have any knowledge of this process.

**Comment by Omico** [comment]
> but the `https://services.gradle.org/distributions/gradle-[VERSION]-wrapper.jar.sha256` doesn't exist yet.

It exists. I can download it manually when I face this issue, but I tried multiple times, and the CI always failed. https://services.gradle.org/distributions/gradle-8.7-rc-3-wrapper.jar.sha256

I circumvented this problem by hardcoding the hash.

```
      - uses: gradle/wrapper-validation-action@v2
        with:
          allow-checksums: 'cb0da6751c2b753a16ac168bb354870ebb1e162e9083f116729cec9c781156b8'

```

I removed the `allow-checksums` and tried `8.7-rc-4`, which works. Then I tried `8.7-rc-3` again, which also works. 

Now, the issue cannot be reproduced and is becoming a mystery. 🫠
